diff --git a/src/vs/platform/extensionManagement/common/extensionGalleryService.ts b/src/vs/platform/extensionManagement/common/extensionGalleryService.ts
index 4034954..06da4ed 100644
--- a/src/vs/platform/extensionManagement/common/extensionGalleryService.ts
+++ b/src/vs/platform/extensionManagement/common/extensionGalleryService.ts
@@ -1939,11 +1939,23 @@ export abstract class AbstractExtensionGalleryService implements IExtensionGalle
 			return { malicious: [], deprecated: {}, search: [], autoUpdate: {} };
 		}
 
-		const context = await this.requestService.request({
+		const excludeUnsafes = this.configurationService.getValue('extensions.excludeUnsafes') ?? true;
+
+		if (!excludeUnsafes) {
+			return { malicious: [], deprecated: {}, search: [] };
+		}
+
+		const requestOptions: IRequestOptions = {
 			type: 'GET',
 			url: this.extensionsControlUrl,
 			timeout: this.getRequestTimeout()
-		}, CancellationToken.None);
+		};
+
+		if (!isWeb) {
+			requestOptions.headers = { Connection: 'close' };
+		}
+
+		const context = await this.requestService.request(requestOptions, CancellationToken.None);
 
 		if (context.res.statusCode !== 200) {
 			throw new Error('Could not get extensions report.');
diff --git a/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts b/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts
index 6956d79..4555c85 100644
--- a/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts
+++ b/src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts
@@ -309,6 +309,13 @@ Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Configuration)
 				scope: ConfigurationScope.APPLICATION,
 				tags: ['advanced', 'usesOnlineServices']
 			},
+			'extensions.excludeUnsafes': {
+				scope: ConfigurationScope.MACHINE,
+				type: 'boolean',
+				default: true,
+				description: localize('extensionsExcludeUnsafes', "When enabled, it will download and cache the list of malicious and deprecated extensions. It's recommended to leave it enabled."),
+				tags: ['usesOnlineServices']
+			},
 		}
 	});
 
