diff --git a/src/vs/workbench/browser/parts/paneCompositeBar.ts b/src/vs/workbench/browser/parts/paneCompositeBar.ts
index b26b6c5..022717d 100644
--- a/src/vs/workbench/browser/parts/paneCompositeBar.ts
+++ b/src/vs/workbench/browser/parts/paneCompositeBar.ts
@@ -340,10 +340,14 @@ export class PaneCompositeBar extends Disposable {
 		for (const viewContainer of viewContainers) {
 			this.addComposite(viewContainer);
 
-			// Pin it by default if it is new
+			// Resonance: Only pin Explorer by default, unpin all other sidebar viewlets
 			const cachedViewContainer = this.cachedViewContainers.filter(({ id }) => id === viewContainer.id)[0];
 			if (!cachedViewContainer) {
-				this.compositeBar.pin(viewContainer.id);
+				if (this.location === ViewContainerLocation.Sidebar && viewContainer.id !== 'workbench.view.explorer') {
+					this.compositeBar.unpin(viewContainer.id);
+				} else {
+					this.compositeBar.pin(viewContainer.id);
+				}
 			}
 
 			// Active
